crates/editor/src/editor.rs

```diff
@@ -7287,17 +7287,68 @@ impl Editor {
         let buffer = self.buffer.read(cx);
         let head = self.selections.newest::<usize>(cx).head();
         let (buffer, head) = buffer.text_anchor_for_position(head, cx)?;
+        let head_offset = text::ToOffset::to_offset(&head, buffer.read(cx));
         let replica_id = self.replica_id(cx);
 
         let workspace = self.workspace()?;
         let project = workspace.read(cx).project().clone();
         let references = project.update(cx, |project, cx| project.references(&buffer, head, cx));
-        Some(cx.spawn(|_, mut cx| async move {
-            let locations = references.await?;
+        Some(cx.spawn(|editor, mut cx| async move {
+            let mut locations = references.await?;
+
+            // Lsp may return references that contain the item itself we requested `find_all_references` for (eg. rust-analyzer)
+            // So we remove it from locations
+            cx.update(|cx| {
+                locations = locations
+                    .iter()
+                    .filter(|r| {
+                        // fn foo(x : i64) {
+                        //         ^
+                        //  println!(x);
+                        // }
+                        // It is ok to find reference when caret being at ^ (the end of the word)
+                        // So we turn offset into inclusive to include the end of the word
+                        !r.range
+                            .to_offset(r.buffer.read(cx))
+                            .to_inclusive()
+                            .contains(&head_offset)
+                    })
+                    .map(|x| x.clone())
+                    .collect();
+            })?;
+
             if locations.is_empty() {
```

Comment:
Current Zed opened multibuffer for such example, showing two items in it:
```rust
fn foo(param: i64) {
    let a = param;
    let a<|> = param;
    let a = param;
    println!(param);
}
```

I've expected it to toggle now, but now `locations.is_empty()` for such piece of code, this is a regression.
Let's add another test and fix it?